internal/builder/packages.go

```diff
@@ -290,19 +293,43 @@ func signZippedPackage(options BuildOptions, zippedPackagePath string) error {
 	return nil
 }
 
-func copyLicenseTextFile(licensePath string) error {
-	_, err := os.Stat(licensePath)
-	if err == nil {
+// copyLicenseTextFile checks if a license file exists in the package directory.
+// If the file already exists in the package, it does nothing.
+// If the file does not exist in the package, it looks for a license file in the repository root directory.
+// If a license file is found in the repository, it copies it to the package directory.
+func copyLicenseTextFile(repoRoot *os.Root, licensePath string) error {
+	// if the given path exist, skip copying
+	info, err := os.Stat(licensePath)
+	if err == nil && !info.IsDir() {
 		logger.Debug("License file in the package will be used")
 		return nil
 	}
+	// if the given path does not exist, continue
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return fmt.Errorf("can't check license path (%s): %w", licensePath, err)
+	}
+	// if the given path exist but is a directory, return an error
+	if info != nil && info.IsDir() {
+		return fmt.Errorf("license path (%s) is a directory", licensePath)
+	}
+
+	// Ensure licensePath is inside the repoRoot
+	rel, err := filepath.Rel(repoRoot.Name(), licensePath)
+	if err != nil {
+		return fmt.Errorf("failed to get relative path for licensePath (%s) from repoRoot (%s): %w", licensePath, repoRoot.Name(), err)
+	}
+	if strings.HasPrefix(rel, "..") {
+		return fmt.Errorf("licensePath (%s) is outside of the repoRoot (%s)", licensePath, repoRoot.Name())
+	}
```

Comment:
This check probably makes sense to move it to `findRepositoryLicensePath` that already checks about the existence of the license file. So, both the value from environment variable and the one from the code are validated.

Would this be equivalent to run `repoRoot.Stat()` (https://go.dev/blog/osroot#introducing-osroot) ? 

```golang
	_, err := repoRoot.Stat(licensePath)
	if err != nil {
		return fmt.Errorf("stat licensePath (%s) from repoRoot (%s) failed: %w", licensePath, repoRoot.Name(), err)
	}
```

Reply:
i've been experimenting with the root api, and although `os.Stat` is able to accept absolute paths, when root.Stat is used with a param that is an absolute path, although is within root, it doest not detect it as a file within its directories.

I would say, no, is not equivalent. root api would be applicable when using relative paths. 

In this case, we are checking that the given `licensePath` where the file has to be copied, exists within root.

I've been testing to change line 302 where i check the file, to use repoRoot.Stat but the output is not the expected.